src/general/IWorldEffect.cs

```diff
@@ -93,3 +95,167 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
         }
     }
 }
+
+[JSONDynamicTypeAllowed]
+public class GasProductionEffect : IWorldEffect
+{
+    [JsonProperty]
+    private GameWorld targetWorld;
+
+    public GasProductionEffect(GameWorld targetWorld)
+    {
+        this.targetWorld = targetWorld;
+    }
+
+    public void OnRegisterToWorld()
+    {
+    }
+
+    public void OnTimePassed(double elapsed, double totalTimePassed)
+    {
+        List<Compound> gasCompounds = SimulationParameters.Instance.GetGasCompounds();
+
+        foreach (var patch in targetWorld.Map.Patches.Values)
+        {
+            // First, add the constant intake to the patch, since it will influence the available amount for species.
+            // TODO move to own effect
+            AddConstantIntakeToPatch(patch);
+
+            Dictionary<Compound, float> compoundsProduced = new Dictionary<Compound, float>(gasCompounds.Count);
```

Comment:
You seem to be a bit underusing `var` in general, but especially this line is unnecessarily long due to not using `var`.

Reply:
I guess old habits die hard. Fixed.